Feature/makefiles qnx

s-bauer opened the pull request:
- Added .editorconfig to specify tab size to 8
- Refactored Makefile.in into different sections to increase readability & maintainability (The github diff seems to mess up the tabs, view it in the file explorer or locally instead)
- Workaround for missing "wc" utility on QNX. Not sure if we should merge that back into the main clicon repository (https://github.com/s-bauer/cligen/blob/feature/makefiles-qnx/Makefile.in#L241)
- Added linkages (https://github.com/s-bauer/cligen/blob/feature/makefiles-qnx/Makefile.in#L148-L149). Again, not sure if this should be merged back.
- Added QNX specific import: https://github.com/s-bauer/cligen/blob/feature/makefiles-qnx/cligen_cv_internal.h#L50-L52
  Is there a way to check for this in the configure(.ac) script?

olofhagsand commented:
- Concerning Makefile changes, I see a lot of formatting changes. Is this intentional or just a side-effect of the commit? If they are intentional, please describe motivation for change. 
- Also please describe the significant (non-formatting) change to Makefile, it gets lost in the formatting change / qnx changes?
- Concerning .editconfig, clixon uses 4-char tab mode, see https://github.com/clicon/clixon/blob/master/doc/DEVELOP.md#c-style. Maybe .editconfig setting is different, please explain if so.

s-bauer commented:
- Makefile changes are 99% formatting changes. Taking a look at the previous Makefile.in, it was a mix of spaces, tabs and not correctly aligned properties. I've mainly aligned the variables and targets, grouped them into logical sections and added correct `.PHONY` statements where missing. The GitHub change view seems to be a bit broken and does not use the 4 char tab width from the editorconfig. The normal file view (https://github.com/clicon/cligen/blob/7146c27fcd30f773632153a718d145c62e36aab3/Makefile.in) does pick it up and correctly shows everything with 4 char tab width.
- The only Makefile changes except formatting are:
    - Added linkage to build target (to create *.so and *.so.MAJOR linked to *.so.MAJOR.MINOR)
    https://github.com/clicon/cligen/blob/7146c27fcd30f773632153a718d145c62e36aab3/Makefile.in#L148-L149
    - On QNX, the "wc" utility does not exist which breaks the determination of the `CLIGEN_BUILDSTR` length. As a workaround we're using `| tr -c '\n' '[\n*]' | grep -c '^'` instead. That workaround should work fine on Linux as well, so it might be an option to commit this into the cligen main. If you don't like this, we can also remove it or put it behind a flag for QNX.
    https://github.com/clicon/cligen/blob/7146c27fcd30f773632153a718d145c62e36aab3/Makefile.in#L240
- Editorconfig: Oh, I didn't see the contribution guide. Judging from the existing clixon source code, which is often using 8 char tab width, I just continued to use it. Maybe you should copy the editorconfig over to the clixon project, so that GitHub correctly displays everything with 4 char tab width.

olofhagsand commented:
When I do make clean; make nothing is built, instead it is built when I do sudo make install which makes all object files be "root".
```
olof@alarik> make clean
rm -f cligen_hello cligen_file cligen_tutorial cligen_object.o cligen_parsetree.o cligen_pt_head.o cligen_handle.o cligen_cv.o cligen_match.o cligen_read.o cligen_io.o cligen_expand.o cligen_syntax.o cligen_print.o cligen_cvec.o cligen_buf.o cligen_util.o cligen_history.o cligen_regex.o cligen_getline.o build.o  lex.cligen_parse.o cligen_parse.tab.o  
rm -f libcligen.so.5.4 libcligen.so.5 libcligen.so 
rm -f *.tab.c *.tab.h *.tab.o 
rm -f lex.*.c lex.*.o
rm -rf cligen
rm -f *.gcda *.gcno *.gcov # coverage
olof@alarik> make
cd . && rm -rf cligen && mkdir cligen && find . -name '*.h' -exec cp --parents '{}' cligen \;
# Nothing is built
```
Instead it is all built.
```
olof@alarik> sudo make install
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_object.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_parsetree.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_pt_head.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_handle.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_cv.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_match.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_read.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_io.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_expand.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_syntax.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_print.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_cvec.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_buf.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_util.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_history.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_regex.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_getline.c
echo "/* This file is generated from the CLIgen Makefile */" > build.c;
date +"const char CLIGEN_BUILDSTR[39]=\"%Y.%m.%d %H:%M by `whoami` on `hostname`"\"\; >> build.c;
echo "const char CLIGEN_VERSION[64]=\"5.4.0.PRE\""\; >> build.c;
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c build.c
bison -y -l -d -b cligen_parse -p cligen_parse ./cligen_parse.y # -t is debug
./cligen_parse.y: warning: 2 shift/reduce conflicts [-Wconflicts-sr]
flex -Pcligen_parse ./cligen_parse.l # -d is debug 
gcc -I. -I.  -DYY_NO_INPUT -fPIC -DHAVE_CONFIG_H  -g -Wall -c lex.cligen_parse.c
gcc -I. -I.  -fPIC -DHAVE_CONFIG_H  -g -Wall -c cligen_parse.tab.c
gcc -fPIC -DHAVE_CONFIG_H  -g -Wall -shared -o libcligen.so.5.4 cligen_object.o cligen_parsetree.o cligen_pt_head.o cligen_handle.o cligen_cv.o cligen_match.o cligen_read.o cligen_io.o cligen_expand.o cligen_syntax.o cligen_print.o cligen_cvec.o cligen_buf.o cligen_util.o cligen_history.o cligen_regex.o cligen_getline.o build.o  lex.cligen_parse.o cligen_parse.tab.o  -Wl,-soname=libcligen.so.5.4 -L.
ln -sf libcligen.so.5.4 libcligen.so.5
ln -sf libcligen.so.5 libcligen.so
/usr/bin/install -c -m 0755 -d /usr/local/lib
/usr/bin/install -c -m 0644  libcligen.so.5.4 /usr/local/lib
(cd /usr/local/lib && ln -s -f libcligen.so.5.4 libcligen.so.5)     # -l:libcligen.so.3
(cd /usr/local/lib && ln -s -f libcligen.so.5 libcligen.so) # -l:libcligen.so
install -m 0755 -d /usr/local/include/cligen
/usr/bin/install -c ./cligen_cv.h ./cligen_cvec.h ./cligen_object.h ./cligen_handle.h ./cligen_parsetree.h ./cligen_pt_head.h ./cligen_print.h ./cligen_read.h ./cligen_io.h ./cligen_expand.h ./cligen_syntax.h ./cligen_buf.h ./cligen_util.h ./cligen_history.h ./cligen_regex.h ./cligen.h /usr/local/include/cligen
olof@alarik> 
```
